src/KUSignalLib/DSP/DSP.py

```diff
@@ -25,4 +26,114 @@ def IIRDirectForm2(b, a, x):
         tmp = np.dot(denominator, delayLine) #df2 left side
         delayLine[1:] = delayLine[:-1] #shift delay line
         delayLine[0] = x[i]*a[0] + tmp #new value is x[n] * a[0] + sum of left side
-    return y
\ No newline at end of file
+    return y
+
+def Interpolate(x, n, mode="linear"):
+    """
+    Perform interpolation on an upsampled signal.
+
+    :param x: Input signal.
+    :param n: Upsampled factor (signal is already upsampled)
+    :param mode: Interpolation type -> "linear", "quadratic"
+    :return: Interpolated signal.
+    """
+    nonzero_indices = np.arange(0, len(x)*n, n) # Generate indices for upsampled signal
+    interpolation_function = intp.interp1d(nonzero_indices, np.array(x), kind=mode, fill_value='extrapolate') # create interpolation function
+    interpolated_signal = interpolation_function(np.arange(len(x)*n)) # interpolate the signal
+    return interpolated_signal
+
+
+def Upsample(x, n, dim=1, interpolate=True, axis=0):
```

Comment:
have the function auto detect the dimension of the input, dont have them specify